### PR TITLE
Converted travis to github ci

### LINKED ISSUE #729: Transition CI from Travis to Github Actions
Part of the general migration from Travis.  Also requires fix for recent Neo4j install failure on Travis.  Jim, going to assign since we spoke, hope that's okay?  Thanks!

### PR DESCRIPTION
**Pull request by jim-hill-r**
Fixes #729

Still needs
- [ ] Testing in github to validate it works
- [x] Updated badging

### CODE REVIEW

**Review by dehann — APPROVED** [review]
oh cool thanks! I don't quite know how to read Actions workflows in detail yet, but looks right from what I know.  (I'm still at level of just copying parts over and see if it works).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      
+      - name: Git Test Credentials
+        run: |
+          git config --global user.name Tester
+          git config --global user.email te@st.er
+      
+      - name: Run tests
+        uses: julia-actions/julia-runtest@latest
+        env:
+          IIF_TEST: false
```
Comment (dehann) [inline .github/workflows/ci.yml:58]: this guy should be for enable disabling test against branch in IIF `dfg_integration_test`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          SKIP_CGDFG_TESTS: true
```
Comment (dehann) [inline .github/workflows/ci.yml:59]: Cool, simple ENV to skip CloudGraphsDFG tests

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          - name: Checkout
+            uses: actions/checkout@v2
+
+          - name: Setup julia
+            uses: julia-actions/setup-julia@v1
+            with:
+              version: 1.4
+              arch: x64
+
+          - name: Install neo4j
```
Comment (dehann) [inline .github/workflows/ci.yml:86]: this was giving issues on Travis, will be cool if it just works again here on Actions

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+          - name: Code Coverage
+            uses: codecov/codecov-action@v1
+            with:
+              file: lcov.info
+            # run: |
+            #   julia -e 'using Pkg; cd(Pkg.dir("DistributedFactorGraphs")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
+
+  docs:
+    needs: test
+    name: Documentation
```
Comment (dehann) [inline .github/workflows/ci.yml:139]: Great, forgot to mention docs in my longer issue comment.  I'm not quite sure how auth tokens for publishing docs works on Actions yet.  Figured it out for Travis a while back.  For references, travis was running docs (Documenter.jl) which does all the magic when on master branch to push new docs to branch `ph-pages` (thats all automated).  Github then hosts that.  Documenter.jl is pretty awesome.

Comment (dehann) [reply .github/workflows/ci.yml:139]: here is an example https://travis-ci.org/github/JuliaRobotics/DistributedFactorGraphs.jl/jobs/752440905

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+      - name: Setup julia
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: 1.4
+          arch: x64
+
+      - name: Documentation
+          run: |
+            julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
+            julia --project=docs/ docs/make.jl
```
Comment (dehann) [inline .github/workflows/ci.yml:155]: yep, I think this should be enough to deploy docs from inside the Julia execution.  All Documenter.jl magic, as long as the access tokens are set up right for Actions to push to this repo.  This part I'm not sure about, but can add that this was a secret env variable on Travis with a simple setup.  Actions likely easier.

**Comment by Affie** [comment]
@jim-hill-r, docs needs a key:
```
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```
see: https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#GitHub-Actions

**Comment by dehann** [comment]
Hi @jim-hill-r , 

Just wanted to check if this was blocked or still in progress, don't want to forget about it.

Best,
Dehann

**Comment by dehann** [comment]
don't merge, im trying to get  the CI running with Actions

**Pull request merged**